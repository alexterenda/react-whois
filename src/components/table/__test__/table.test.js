import React from 'react';
import renderer from 'react-test-renderer';

import Table from '../table';

const testDomainData = [
  {
    key: 'address',
    name: 'Address',
    value: '10 Corporate Drive Suite 300',
  },
  {
    key: 'city',
    name: 'City',
    value: 'Burlington',
  },
  {
    key: 'country',
    name: 'Country',
    value: 'US',
  },
  {
    key: 'creation_date',
    name: 'Creation Date',
    value: 'Fri, 01 Jul 1994 00:00:00 GMT',
  },
  {
    key: 'creation_date ',
    name: 'Creation Date',
    value: 'Fri, 01 Jul 1994 04:00:00 GMT',
  },
  {
    key: 'dnssec',
    name: 'Dnssec',
    value: 'unsigned',
  },
  {
    key: 'domain_name',
    name: 'Domain Name',
    value: 'DOMAIN.COM',
  },
  {
    key: 'emails',
    name: ' Emails',
    value: 'dtransfers@registerapi.com',
  },
  {
    key: 'emails ',
    name: ' Emails',
    value: 'jason@endurance.com',
  },
  {
    key: 'emails ',
    name: ' Emails',
    value: 'mike.chattan@endurance.com',
  },
  {
    key: 'emails ',
    name: ' Emails',
    value: 'mark@endurance.com',
  },
  {
    key: 'emails ',
    name: ' Emails',
    value: 'linda.scott@endurance.com',
  },
  {
    key: 'emails ',
    name: ' Emails',
    value: 'joanna.krith@endurance.com',
  },
  {
    key: 'emails ',
    name: ' Emails',
    value: 'compliance@domain-inc.net',
  },
  {
    key: 'emails ',
    name: ' Emails',
    value: 'corpdomains@endurance.com',
  },
  {
    key: 'expiration_date',
    name: 'Expiration Date',
    value: 'Thu, 07 Jan 2021 00:00:00 GMT',
  },
  {
    key: 'expiration_date ',
    name: 'Expiration Date',
    value: 'Thu, 07 Jan 2021 13:34:24 GMT',
  },
  {
    key: 'name',
    name: ' Name',
    value: 'Domain Administrator',
  },
  {
    key: 'name_servers',
    name: ' Name Servers',
    value: 'NS-1250.AWSDNS-28.ORG',
  },
  {
    key: 'name_servers ',
    name: ' Name Servers',
    value: 'NS-166.AWSDNS-20.COM',
  },
  {
    key: 'name_servers ',
    name: ' Name Servers',
    value: 'NS-2022.AWSDNS-60.CO.UK',
  },
  {
    key: 'name_servers ',
    name: ' Name Servers',
    value: 'NS-683.AWSDNS-21.NET',
  },
  {
    key: 'org',
    name: ' Org',
    value: 'Endurance International Group West, Inc',
  },
  {
    key: 'referral_url',
    name: 'Referral Url',
    value: 'http://www.nic.ae',
  },
  {
    key: 'registrar',
    name: 'Registrar',
    value: 'Domain.com, LLC',
  },
  {
    key: 'state',
    name: 'State',
    value: 'MA',
  },
  {
    key: 'status',
    name: 'Status',
    value:
      'clientDeleteProhibited https://icann.org/epp#clientDeleteProhibited',
  },
  {
    key: 'status ',
    name: 'Status',
    value:
      'clientTransferProhibited https://icann.org/epp#clientTransferProhibited',
  },
  {
    key: 'status ',
    name: 'Status',
    value:
      'clientUpdateProhibited https://icann.org/epp#clientUpdateProhibited',
  },
  {
    key: 'updated_date',
    name: 'Updated Date',
    value: 'Fri, 21 Oct 2016 00:00:00 GMT',
  },
  {
    key: 'updated_date ',
    name: 'Updated Date',
    value: 'Fri, 21 Oct 2016 20:11:05 GMT',
  },
  {
    key: 'whois_server',
    name: 'Whois Server',
    value: 'whois.nic.ae',
  },
  {
    key: 'zipcode',
    name: 'Zipcode',
    value: '01803',
  },
];

it('matches snapshot', () => {
  const tree = renderer.create(<Table domainData={testDomainData} />).toJSON();
  expect(tree).toMatchSnapshot();
});
