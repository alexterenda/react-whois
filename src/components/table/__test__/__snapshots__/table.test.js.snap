// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<table
  data-testid="table"
>
  <thead>
    <tr>
      <th>
        Record
      </th>
      <th>
        Value
      </th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td>
        Address
      </td>
      <td>
        10 Corporate Drive Suite 300
      </td>
    </tr>
    <tr>
      <td>
        City
      </td>
      <td>
        Burlington
      </td>
    </tr>
    <tr>
      <td>
        Country
      </td>
      <td>
        US
      </td>
    </tr>
    <tr>
      <td>
        Creation Date
      </td>
      <td>
        Fri, 01 Jul 1994 00:00:00 GMT
      </td>
    </tr>
    <tr>
      <td>
        Creation Date
      </td>
      <td>
        Fri, 01 Jul 1994 04:00:00 GMT
      </td>
    </tr>
    <tr>
      <td>
        Dnssec
      </td>
      <td>
        unsigned
      </td>
    </tr>
    <tr>
      <td>
        Domain Name
      </td>
      <td>
        DOMAIN.COM
      </td>
    </tr>
    <tr>
      <td>
         Emails
      </td>
      <td>
        dtransfers@registerapi.com
      </td>
    </tr>
    <tr>
      <td>
         Emails
      </td>
      <td>
        jason@endurance.com
      </td>
    </tr>
    <tr>
      <td>
         Emails
      </td>
      <td>
        mike.chattan@endurance.com
      </td>
    </tr>
    <tr>
      <td>
         Emails
      </td>
      <td>
        mark@endurance.com
      </td>
    </tr>
    <tr>
      <td>
         Emails
      </td>
      <td>
        linda.scott@endurance.com
      </td>
    </tr>
    <tr>
      <td>
         Emails
      </td>
      <td>
        joanna.krith@endurance.com
      </td>
    </tr>
    <tr>
      <td>
         Emails
      </td>
      <td>
        compliance@domain-inc.net
      </td>
    </tr>
    <tr>
      <td>
         Emails
      </td>
      <td>
        corpdomains@endurance.com
      </td>
    </tr>
    <tr>
      <td>
        Expiration Date
      </td>
      <td>
        Thu, 07 Jan 2021 00:00:00 GMT
      </td>
    </tr>
    <tr>
      <td>
        Expiration Date
      </td>
      <td>
        Thu, 07 Jan 2021 13:34:24 GMT
      </td>
    </tr>
    <tr>
      <td>
         Name
      </td>
      <td>
        Domain Administrator
      </td>
    </tr>
    <tr>
      <td>
         Name Servers
      </td>
      <td>
        NS-1250.AWSDNS-28.ORG
      </td>
    </tr>
    <tr>
      <td>
         Name Servers
      </td>
      <td>
        NS-166.AWSDNS-20.COM
      </td>
    </tr>
    <tr>
      <td>
         Name Servers
      </td>
      <td>
        NS-2022.AWSDNS-60.CO.UK
      </td>
    </tr>
    <tr>
      <td>
         Name Servers
      </td>
      <td>
        NS-683.AWSDNS-21.NET
      </td>
    </tr>
    <tr>
      <td>
         Org
      </td>
      <td>
        Endurance International Group West, Inc
      </td>
    </tr>
    <tr>
      <td>
        Referral Url
      </td>
      <td>
        http://www.nic.ae
      </td>
    </tr>
    <tr>
      <td>
        Registrar
      </td>
      <td>
        Domain.com, LLC
      </td>
    </tr>
    <tr>
      <td>
        State
      </td>
      <td>
        MA
      </td>
    </tr>
    <tr>
      <td>
        Status
      </td>
      <td>
        clientDeleteProhibited https://icann.org/epp#clientDeleteProhibited
      </td>
    </tr>
    <tr>
      <td>
        Status
      </td>
      <td>
        clientTransferProhibited https://icann.org/epp#clientTransferProhibited
      </td>
    </tr>
    <tr>
      <td>
        Status
      </td>
      <td>
        clientUpdateProhibited https://icann.org/epp#clientUpdateProhibited
      </td>
    </tr>
    <tr>
      <td>
        Updated Date
      </td>
      <td>
        Fri, 21 Oct 2016 00:00:00 GMT
      </td>
    </tr>
    <tr>
      <td>
        Updated Date
      </td>
      <td>
        Fri, 21 Oct 2016 20:11:05 GMT
      </td>
    </tr>
    <tr>
      <td>
        Whois Server
      </td>
      <td>
        whois.nic.ae
      </td>
    </tr>
    <tr>
      <td>
        Zipcode
      </td>
      <td>
        01803
      </td>
    </tr>
  </tbody>
</table>
`;
